.github/workflows/pr.yml

```diff
@@ -31,22 +31,47 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        java: [ 8, 11, 17, 21 ]
+        java: [ 21, 8, 11, 17 ]
     name: Compile and Check Java ${{ matrix.java }}
     steps:
       - name: Checkout code
         uses: actions/checkout@v4
         with:
           persist-credentials: false
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
       - name: Setup Gradle
         uses: ./.github/actions/setup-gradle
         with:
           java-version: ${{ matrix.java }}
           gradle-cache-read-only: true
           develocity-access-key: ${{ secrets.GE_ACCESS_TOKEN }}
       - name: Compile and validate
-        # Publish build scans for now. This will only work on PRs from apache/kafka, not public forks.
-        run: ./gradlew --build-cache --info --scan check -x test
+        # Gradle flags
+        # --build-cache:  Let Gradle restore the build cache
+        # --info:         For now, we'll generate lots of logs while setting up the GH Actions
+        # --scan:         Attempt to publish build scans in PRs. This will only work on PRs from apache/kafka, not public forks.
+        # spotlessCheck:  Skip spotless. While it is faster than checkstyle, it does not produce a report
+        #                 which makes it a bit less useful when running on CI.
+        run: ./gradlew --build-cache --info --scan check -x test -x spotlessCheck
```

Comment:
Good to know. I've backed out this change already. Do you know if we are planning on keeping both checkstyle and spotless around for a while? Or are we trying to migrated only to spotless?

Reply:
> Do you know if we are planning on keeping both checkstyle and spotless around for a while? Or are we trying to migrated only to spotless?

Yes, I prefer to leverage spotless as much as possible as it can fix the code automatically (for example, re-format the imports). We will dig in it after dropping JDK8. there is already a related jira https://issues.apache.org/jira/browse/KAFKA-16896